dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/netty/jfapchannel/NettyNetworkConnection.java

```diff
@@ -0,0 +1,330 @@
+	{
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "<init>" ,new Object[] {bootstrap, chainName});
+
+//		init(bootstrap);
+		this.chainName = chainName;
+		this.sslOptions = sslOptions;
+		this.isInbound = isInbound;
+		this.tlsProvider = tlsProvider;
+		this.nettyBundle = nettyBundle;
+		// TODO Check if we need to clone this
+		this.bootstrap = bootstrap;
+
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(tc, "<init>", new Object[] {bootstrap, chainName});
+	}
+
+	public NettyNetworkConnection(Channel chan, String chainName, boolean isInbound)
+	{
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "<init>", chan);
+
+		this.chan = chan;
+		this.isInbound = isInbound;
+		this.chainName = chainName;
+
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(tc, "<init>", new Object[] {chan});
+	}
+
+	public void linkOutboundConnection(OutboundConnection conn) throws NettyException {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "linkOutboundConnection");
+		if(this.chan == null) {
+			throw new NettyException("Error linking connection appropriately");
+		}
+		if(chan.attr(NettyJMSClientHandler.CONNECTION_KEY).get() != null) {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) SibTr.debug(this, tc, "linkOutboundConnection","Connection was already set: "+chan.attr(NettyJMSClientHandler.CONNECTION_KEY).get());
+		}
+		chan.attr(NettyJMSClientHandler.CONNECTION_KEY).set(conn);
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "linkOutboundConnection");
+	}
+
+
+	/**
+	 * @return Returns the channel connection.
+	 */
+	Channel getVirtualConnection()
+	{
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "getVirtualConnection");
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "getVirtualConnection", chan);
+		return this.chan;
+	}
+
+	/**
+	 *
+	 * @see com.ibm.ws.sib.jfapchannel.framework.NetworkConnection#requestPermissionToClose(long)
+	 */
+	public boolean requestPermissionToClose(long timeout)
+	{
+		// TODO Figure out the netty equivalent for this. Only used in connection
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "requestPermissionToClose", Long.valueOf(timeout));
+		//	      boolean canProcess = vc.requestPermissionToClose(timeout);
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "requestPermissionToClose", Boolean.valueOf(true));
+		return true;
+	}
+
+	/**
+	 * Sets the timeout to the values passed
+	 * @param timeout in seconds
+	 * @throws NettyException 
+	 */
+	public void setHearbeatInterval(int timeout) throws NettyException {
```

Comment:
seconds seems like a rather chunky unit to use internally...fine if there is some reason for it.